pkg/project/v2/project.go

```diff
@@ -48,25 +51,29 @@ type Project struct {
 
 	// Configs are the configurations within this Project
 	Configs ConfigsPerTypePerEnvironments
+}
 
-	// Dependencies of this project to other projects
-	Dependencies DependenciesPerEnvironment
+func (p Project) ConfigList() []config.Config {
+	var configs []config.Config
+	for x := range maps.Values(p.Configs) {
+		for y := range maps.Values(x) {
+			configs = append(configs, y...)
+		}
+	}
+	return configs
 }
 
 // HasDependencyOn returns whether the project it is called on, has a dependency on the given project, for the given environment
 func (p Project) HasDependencyOn(environment string, project Project) bool {
-	dependencies, found := p.Dependencies[environment]
-
-	if !found {
-		return false
-	}
-
-	for _, dep := range dependencies {
-		if dep == project.Id {
-			return true
+	for c := range slices.Values(p.ConfigList()) {
```

Comment:
This method just went from a `O(n)` to an `O(n^4)` (`O(n^2)` from ConfigList) + `O(n^2)` from this method). 
Furthermore, in ConfigList we do collect the configs now *on every* call, just to find the dependencies. This is an intended redundency so that other parts of Monaco work in a timely manner. We took much time to optimize this method (and the whole topology sort) to what it is today to optimize it. 

This refactoring here is not acceptable, and must not be done.